# Récupération et traitement des données

In [81]:
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
import geopandas as gpd
import contextily as ctx
import pynsee 
from pynsee.download import download_file
from pynsee.geodata import get_geodata
import requests
from shapely.geometry import Point
from geopy.geocoders import Nominatim
from cartiflette.s3 import download_vectorfile_url_all
from io import BytesIO
from zipfile import ZipFile

In [2]:
from pathlib import Path

In [3]:
Path.cwd()

PosixPath('/home/cathu/Documents/ENSAE/Projet_Catherine_Christelle')

## Les aménagements cyclables en Ile de France

In [4]:
a_velo= gpd.read_file('amenagements-velo-en-ile-de-france.geojson')

In [5]:
# Nombre d'entrées
a_velo.shape[0]

112217

In [6]:
a_velo.crs

<Geographic 2D CRS: EPSG:4326>
Name: WGS 84
Axis Info [ellipsoidal]:
- Lat[north]: Geodetic latitude (degree)
- Lon[east]: Geodetic longitude (degree)
Area of Use:
- name: World.
- bounds: (-180.0, -90.0, 180.0, 90.0)
Datum: World Geodetic System 1984 ensemble
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich

Un préalable au calcul des surfaces est le choix du système de projection adéquat. Dans notre cas , il s'agit de convertir les données au système de projection Lambert 93 qui est le plus approprié.

In [7]:
a_velo['longueur'].describe()

count    112217.000000
mean        126.315959
std         174.519817
min           0.000000
25%          24.000000
50%          70.000000
75%         161.000000
max        3984.000000
Name: longueur, dtype: float64

In [8]:
a_velo.sample(n=1)

osm_id nom_com sens_voit              ag panneaux moyenn_ech  \
20082  1.123832e+09   Cergy        NC  voie verte uni  FR:C115         11   

      revetement highway insee_com nom_voie  longueur petite_ech           nv  \
20082    asphalt    path     95127     None       214          1  hors voirie   

                   ad                                           geometry  
20082  voie verte uni  LINESTRING (2.01301 49.05301, 2.01272 49.05281...

Pour le moment, nous travaillons sans les données de géolocalisation qui ne sont pas nécessaires aux calculs. Nous créons pour cela une nouvelle dataframe

In [9]:
col_to_keep = ["nom_com", "sens_voit", "ag", "panneaux", "revetement", "highway", "insee_com", "longueur", "nv", "ad"] 

In [10]:
# df_a_velo = a_velo[col_to_keep].copy()

df_a_velo = a_velo

print(df_a_velo.head())

         osm_id            nom_com sens_voit                              ag  \
0  4.014620e+08            Chelles    DOUBLE                            None   
1  4.048666e+08   La Queue-en-Brie        NC  chemin service site propre uni   
2  1.154304e+09            Lésigny        NC                  voie verte uni   
3  3.300503e+08  Pontault-Combault    UNIQUE                       DSC bande   
4  1.104033e+09   Champs-sur-Marne    UNIQUE                            None   

  panneaux moyenn_ech revetement      highway insee_com  \
0     None         32       None      service     77108   
1     None         11       None        track     94060   
2     None         11    asphalt         path     77249   
3     None         22       None  residential     77373   
4     None         22    asphalt  residential     77083   

                     nom_voie  longueur petite_ech           nv  \
0  Rue de la Mare Longue Noue        24          3    limite 30   
1             Allée Jacquette 

### Toutes les pistes ne se valent pas : création de variables d'étude pour les aménagements cyclables

Nous souhaitons étudier la répartition géograhique des aménagements cyclables et particulièrement leur densité en mètre par habitants. Toutefois, l'ensemble des pistes cyclables n'est pas de la même "qualité" : séparée de la route ou non, sens inverse de la circulation, type de revêtement, etc. Pour prendre en compte la qualité des aménagements cyclables, nous pouvons donner des coefficients aux mètres de pistes selon ces différents critères lors du calcul du nombre total de mètres aménagés par commune. Nous allons proposer plusieurs méthodes de calcul selon ces critères.

In [11]:
print(df_a_velo["sens_voit"].unique())
print(df_a_velo["ag"].unique())
print(df_a_velo["revetement"].unique())
print(df_a_velo["highway"].unique())
print(df_a_velo["nv"].unique())
print(df_a_velo["ad"].unique())

['DOUBLE' 'NC' 'UNIQUE']
[None 'chemin service site propre uni' 'voie verte uni' 'DSC bande'
 'autre chemin velo uni' 'piste uni' 'DSC' 'piste trottoir uni'
 'bande uni' 'cheminement trottoir uni' 'goulotte' 'chemin dedie uni'
 'chaucidou' 'cheminement uni' 'DSC piste' 'voie bus uni' 'piste bi'
 'bande bi' 'shoulder uni']
[None 'asphalt' 'compacted' 'concrete' 'unpaved' 'wood' 'paving_stones'
 'ground' 'sett' 'gravel' 'sand' 'paved' 'fine_gravel' 'dirt'
 'cobblestone' 'concrete:plates' 'concrete:lanes' 'grass'
 'cobblestone:flattened' 'metal' 'pebblestone' 'unhewn_cobblestone'
 'bricks' 'tartan' 'pavés' 'vegecol' 'artificial_turf' 'grass_paver'
 'earth' 'bitume' 'à_définir' 'rock']
['service' 'track' 'path' 'residential' 'footway' 'cycleway' 'secondary'
 'tertiary' 'living_street' 'unclassified' 'pedestrian' 'steps' 'primary'
 'primary_link' 'secondary_link' 'tertiary_link' 'trunk_link'
 'motorway_link']
['limite 30' 'hors voirie' None 'z20' 'rue pietonne' 'z30' 'escalier velo'
 'velor

In [12]:
df_a_velo['sens_voit'].isna().sum()

0

In [13]:
df_a_velo['ag'].isna().sum()

68312

In [14]:
df_a_velo['revetement'].isna().sum()

38711

In [15]:
df_a_velo['highway'].isna().sum()

0

In [16]:
df_a_velo['nv'].isna().sum()

9548

In [17]:
df_a_velo['ad'].isna().sum()

68522

In [18]:
print(df_a_velo['highway'].value_counts().get('unclassified', 0))

4025


In [19]:
df_a_velo['adg'] = df_a_velo.apply(lambda row: row['ad'] if pd.notna(row['ad']) else (row['ag'] if pd.notna(row['ag']) else 'unclassified'), axis=1)

In [20]:
print(df_a_velo['adg'].value_counts().get('unclassified', 0))

55496


Seul le critère suivant ne présente pas trop de valeurs manquantes (nommées 'unclassified' dans la base) :  type de route ("highway", cf [documentation OpenstreeMap](https://wiki.openstreetmap.org/wiki/Key:highway) ). Les autres variables ont soit trop de valeurs manquantes soit manquent d'intérêt seule (sens des voitures). Cependant, "highway" est une colonne générale de catégorisation pour les données d'OpenStreetMap et manque de spécificité pour l'étude des pistes cyclables. Nous allons donc proposer deux méthodes de pondération (en plus d'une variable non pondérée) : une avec "highway", une avec "adg" (['nature de la voie'](https://opendata.stif.info/api/datasets/1.0/amenagements-velo-en-ile-de-france/attachments/metadonnees_amenagements_velo_en_ile_de_france_pdf/)), en proposant de normer les chemins dont nous n'avons pas connaissance de la qualité. Le nombre de valeurs manquantes est cependant très important, malgré notre tentative de réduire son ampleur en sommant ad et ag (ce sont les types de voies à gauche et à droite). Roulant à droite en, nous avons donner la priorité à voie à droite.

Nous allons produire 3 variables : deux pondérées censées prendre en compte la qualité de la route, et une autre sans pondération, pour la longueur d'aménagements cyclables par ville. Pour prendre en compte la qualité nous trions les types de routes en leur affectant des poids selon leurs caractéristiques : bande le long d'une route ou séparation, chemin avec ou sans voiture, avec ou sans piéton, etc. Pour cela, la variable "adg" nous semble plus adaptée car plus précise, mais elle a le défaut de présenter énormément valeurs manquantes. Lorsque la valeur est manquante nous assignons un poids neutre (1). La comparaison des résultats avec les 2 méthodes de pondération ou la méthode sans pondération pourra aussi nous renseigner sur la qualité de notre catégorisation.

In [21]:
print(df_a_velo["adg"].unique())

['unclassified' 'chemin service site propre uni' 'voie verte uni'
 'DSC bande' 'bande uni' 'autre chemin velo uni' 'piste uni' 'DSC'
 'piste trottoir uni' 'cheminement trottoir uni' 'goulotte'
 'chemin dedie uni' 'chaucidou' 'cheminement uni' 'voie bus uni'
 'DSC piste' 'piste bi' 'bande bi' 'shoulder uni']


In [22]:
highway_quality_mapping = {
    'service': 1,
    'track': 1,
    'path': 1,
    'trunk_link': 1,
    'motorway_link': 1,
    'residential': 2,
    'footway': 2,
    'cycleway': 2,
    'primary': 2,    
    'primary_link': 2,
    'unclassified': 2,
    'secondary': 3,
    'tertiary': 3,
    'secondary_link': 3,
    'tertiary_link': 3,
    'living_street': 3,
    'pedestrian': 4,
    'steps': 4
}


adg_quality_mapping = {
    'chemin service site propre uni': 1,
    'chemin dedie uni': 1,
    'voie verte uni': 1,
    'autre chemin velo uni': 1,
    'piste bi' : 1,
    'piste uni': 1,
    'bande uni': 2,
    'bande bi' : 2,
    'unclassified': 2,
    'cheminement trottoir uni' : 3,
    'piste trottoir uni': 3,
    'chaucidou': 3,
    'cheminement uni': 3,
    'shoulder uni' : 3,
    'voie bus uni': 4,
    'goulotte': 4,
    'DSC': 4,
    'DSC bande' : 4,
    'DSC piste' : 4
}

In [23]:
df_a_velo['qual_hw'] = df_a_velo['highway'].map(highway_quality_mapping)

df_a_velo['qual_adg'] = df_a_velo['adg'].map(adg_quality_mapping)

In [24]:
quality_weights = {
    1: 1.25,
    2: 1.,
    3: 0.75,
    4: 0.5
}

In [25]:
# Nouvelle colonne pour la longueur pondérée highway

df_a_velo['longueur_pond_hw'] = df_a_velo['longueur'] * df_a_velo['qual_hw'].map(quality_weights)

# Nouvelle colonne pour la longueur pondérée nature voie

df_a_velo['longueur_pond_adg'] = df_a_velo['longueur'] * df_a_velo['qual_adg'].map(quality_weights)

In [26]:
total_longueur_pond_hw = df_a_velo['longueur_pond_hw'].sum()
print(total_longueur_pond_hw) #longueur totale de l'échantillon avec majoration pondérée

14028145.5


In [27]:
total_longueur_pond_nv = df_a_velo['longueur_pond_adg'].sum()
print(total_longueur_pond_nv) #longueur totale de l'échantillon avec majoration pondérée

14382204.5


In [28]:
total_longueur = df_a_velo['longueur'].sum()
print(total_longueur)

14174798


In [29]:
total_longueur_commune = df_a_velo.groupby(['nom_com', 'insee_com'])['longueur'].sum().reset_index()

In [30]:
total_longueur_pond_hw_commune = df_a_velo.groupby(['nom_com', 'insee_com'])['longueur_pond_hw'].sum().reset_index()

In [31]:
total_longueur_pond_adg_commune = df_a_velo.groupby(['nom_com', 'insee_com'])['longueur_pond_adg'].sum().reset_index()

In [32]:
df_amenagements = pd.merge(total_longueur_commune, total_longueur_pond_hw_commune, on=['nom_com', 'insee_com'], suffixes=('_non_pond', '_pond_hw'))

In [33]:
df_amenagements = pd.merge(df_amenagements, total_longueur_pond_adg_commune, on=['nom_com', 'insee_com'], suffixes=(None, '_pond_adg'))

In [34]:
df_amenagements.sample(10)

nom_com insee_com  longueur  longueur_pond_hw  \
469           Le Port-Marly     78502      4390           4568.50   
58                Auvernaux     91037      1666           1852.75   
449     Le Châtelet-en-Brie     77100      7303           7459.75   
882      Tremblay-en-France     93073     35228          36772.00   
598  Montigny-le-Bretonneux     78423     81048          80518.25   
170        Chalou-Moulineux     91131      3619           3530.25   
718      Puiselet-le-Marais     91508       422            316.50   
873                 Tigeaux     77466        30             30.00   
715      Prunay-en-Yvelines     78506       275            343.75   
116                 Bouafle     78090     14898          14709.75   

     longueur_pond_adg  
469            5094.25  
58             1897.75  
449            7516.50  
882           39370.50  
598           91675.50  
170            3926.00  
718             422.00  
873              37.50  
715             310.25  
116           15163.50

In [35]:
df_amenagements['difference_hw'] = df_amenagements['longueur_pond_hw'] - df_amenagements['longueur']
df_amenagements['difference_adg'] = df_amenagements['longueur_pond_adg'] - df_amenagements['longueur']

In [114]:
df_amenagements.head(10)

nom_com insee_com  longueur  longueur_pond_hw  \
0  10e Arrondissement     75110     60281          55764.25   
1  11e Arrondissement     75111     90116          84508.75   
2  12e Arrondissement     75112    201748         201066.75   
3  13e Arrondissement     75113    137731         128770.50   
4  14e Arrondissement     75114    102618          94183.00   
5  15e Arrondissement     75115    148909         136007.00   
6  16e Arrondissement     75116    228240         226273.25   
7  17e Arrondissement     75117    114273         108548.00   
8  18e Arrondissement     75118    118171         113091.00   
9  19e Arrondissement     75119    133478         125675.25   

   longueur_pond_adg  difference_hw  difference_adg  
0           51170.00       -4516.75        -9111.00  
1           73843.00       -5607.25       -16273.00  
2          202496.25        -681.25          748.25  
3          119280.50       -8960.50       -18450.50  
4           87486.75       -8435.00       -15131.25  
5          128724.50      -12902.00       -20184.50  
6          231651.00       -1966.75         3411.00  
7          100289.50       -5725.00       -13983.50  
8           98709.50       -5080.00       -19461.50  
9          128372.00       -7802.75        -5106.00

Nous avons créé nos trois indicateurs par commune : longueur et longueur pondérée (adg et hw) par commune, mais nous avons perdu de l'information dans cette opération puisque nous perdons de ce fait les coordonnées exactes des aménagements. Cependant, pour l'étude de la densité, cela pourra aussi s'avérer utile. Surtout nous avons besoin de ces indicateurs pour le travail économétrique que nous souhaitons mener, où nous considérons alors les communes comme des individus, et la longueur des pistes cyclables comme nous variable d'intérêt. Cependant, avant de passer à la modélisation économétrique pour tenter d'expliquer le développement des aménagements dans les différentes communes, nous souhaitons représenter et analyser spatialement la répartition des aménagements, leur densité (par habitant), leur densité (par kilomètre) et notamment compléter ces analyses grâce à la disponibilité des données du Vélib.

## Données et ressources nécessaires pour l'étude spatiale

In [67]:
communes = download_vectorfile_url_all(
    crs = 4326,
    borders="COMMUNE_ARRONDISSEMENT",
    values = ["75","77","78","91", "92", "93", "94","95"],
    vectorfile_format="topojson",
    filter_by="DEPARTEMENT",
    source="EXPRESS-COG-CARTO-TERRITOIRE",
    year=2022)

https://minio.lab.sspcloud.fr/projet-cartiflette/diffusion/shapefiles-test1/year=2022/administrative_level=COMMUNE_ARRONDISSEMENT/crs=4326/DEPARTEMENT=75/vectorfile_format=topojson/provider=IGN/source=EXPRESS-COG-CARTO-TERRITOIRE/raw.topojson


Downloading: : 40.1kiB [00:00, 1.05MiB/s]


https://minio.lab.sspcloud.fr/projet-cartiflette/diffusion/shapefiles-test1/year=2022/administrative_level=COMMUNE_ARRONDISSEMENT/crs=4326/DEPARTEMENT=77/vectorfile_format=topojson/provider=IGN/source=EXPRESS-COG-CARTO-TERRITOIRE/raw.topojson


Downloading: : 1.57MiB [00:01, 1.01MiB/s]


https://minio.lab.sspcloud.fr/projet-cartiflette/diffusion/shapefiles-test1/year=2022/administrative_level=COMMUNE_ARRONDISSEMENT/crs=4326/DEPARTEMENT=78/vectorfile_format=topojson/provider=IGN/source=EXPRESS-COG-CARTO-TERRITOIRE/raw.topojson


Downloading: : 581kiB [00:00, 827kiB/s] 


https://minio.lab.sspcloud.fr/projet-cartiflette/diffusion/shapefiles-test1/year=2022/administrative_level=COMMUNE_ARRONDISSEMENT/crs=4326/DEPARTEMENT=91/vectorfile_format=topojson/provider=IGN/source=EXPRESS-COG-CARTO-TERRITOIRE/raw.topojson


Downloading: : 483kiB [00:00, 960kiB/s] 


https://minio.lab.sspcloud.fr/projet-cartiflette/diffusion/shapefiles-test1/year=2022/administrative_level=COMMUNE_ARRONDISSEMENT/crs=4326/DEPARTEMENT=92/vectorfile_format=topojson/provider=IGN/source=EXPRESS-COG-CARTO-TERRITOIRE/raw.topojson


Downloading: : 63.0kiB [00:00, 1.17MiB/s]


https://minio.lab.sspcloud.fr/projet-cartiflette/diffusion/shapefiles-test1/year=2022/administrative_level=COMMUNE_ARRONDISSEMENT/crs=4326/DEPARTEMENT=93/vectorfile_format=topojson/provider=IGN/source=EXPRESS-COG-CARTO-TERRITOIRE/raw.topojson


Downloading: : 120kiB [00:00, 862kiB/s]


https://minio.lab.sspcloud.fr/projet-cartiflette/diffusion/shapefiles-test1/year=2022/administrative_level=COMMUNE_ARRONDISSEMENT/crs=4326/DEPARTEMENT=94/vectorfile_format=topojson/provider=IGN/source=EXPRESS-COG-CARTO-TERRITOIRE/raw.topojson


Downloading: : 111kiB [00:00, 978kiB/s] 


https://minio.lab.sspcloud.fr/projet-cartiflette/diffusion/shapefiles-test1/year=2022/administrative_level=COMMUNE_ARRONDISSEMENT/crs=4326/DEPARTEMENT=95/vectorfile_format=topojson/provider=IGN/source=EXPRESS-COG-CARTO-TERRITOIRE/raw.topojson


Downloading: : 372kiB [00:00, 1.00MiB/s]


Un préalable au calcul des surfaces est le choix du système de projection adéquat. Dans notre cas , il s'agit de convertir les données au système de projection Lambert 93 qui est le plus approprié.

In [38]:
communes['surface'] = communes.to_crs(2154).area

In [39]:
communes.sort_values('surface', ascending = False)

id    ID                    NOM  \
111  COMMUNE_0000000009739723  None          Fontainebleau   
148  COMMUNE_0000002200276627  None  Saint-Germain-en-Laye   
135  COMMUNE_0000002200276622  None     Chenoise-Cucharmoy   
229  COMMUNE_0000000009738401  None               Sonchamp   
152  COMMUNE_0000000009739738  None                Étampes   
..                        ...   ...                    ...   
13   COMMUNE_0000000009736527  None            Saint-Mandé   
155  COMMUNE_0000000009734016  None             Gouzangrez   
181  COMMUNE_0000000009735023  None               Margency   
11   COMMUNE_0000000009736027  None   Le Pré-Saint-Gervais   
20   COMMUNE_0000000009735012  None            Vaudherland   

                     NOM_M INSEE_COM           STATUT  POPULATION INSEE_CAN  \
111          FONTAINEBLEAU     77186  Sous-préfecture       15696        07   
148  SAINT-GERMAIN-EN-LAYE     78551  Sous-préfecture       44806        16   
135     CHENOISE-CUCHARMOY     77109   Commune simple        1640        18   
229               SONCHAMP     78601   Commune simple        1600        14   
152                ETAMPES     91223  Sous-préfecture       25629        08   
..                     ...       ...              ...         ...       ...   
13             SAINT-MANDE     94067   Commune simple       22377        23   
155             GOUZANGREZ     95282   Commune simple         160        16   
181               MARGENCY     95369   Commune simple        2846        15   
11    LE PRE-SAINT-GERVAIS     93061   Commune simple       17351        15   
20             VAUDHERLAND     95633   Commune simple          98        21   

    INSEE_ARR INSEE_DEP INSEE_REG           SIREN_EPCI  \
111         4        77        11            200072346   
148         3        78        11            200058519   
135         3        77        11            200037133   
229         2        78        11            200073344   
152         1        91        11            200017846   
..        ...       ...       ...                  ...   
13          2        94        11  200054781/200057941   
155         3        95        11            200035970   
181         2        95        11            200056380   
11          1        93        11  200054781/200057875   
20          2        95        11            200055655   

                               source INSEE_COG  \
111  IGN:EXPRESS-COG-CARTO-TERRITOIRE     77186   
148  IGN:EXPRESS-COG-CARTO-TERRITOIRE     78551   
135  IGN:EXPRESS-COG-CARTO-TERRITOIRE     77109   
229  IGN:EXPRESS-COG-CARTO-TERRITOIRE     78601   
152  IGN:EXPRESS-COG-CARTO-TERRITOIRE     91223   
..                                ...       ...   
13   IGN:EXPRESS-COG-CARTO-TERRITOIRE     94067   
155  IGN:EXPRESS-COG-CARTO-TERRITOIRE     95282   
181  IGN:EXPRESS-COG-CARTO-TERRITOIRE     95369   
11   IGN:EXPRESS-COG-CARTO-TERRITOIRE     93061   
20   IGN:EXPRESS-COG-CARTO-TERRITOIRE     95633   

                                              geometry       surface  
111  POLYGON ((2.74740 48.48329, 2.74783 48.48327, ...  1.722864e+08  
148  POLYGON ((2.07077 48.94649, 2.07138 48.94676, ...  5.313401e+07  
135  POLYGON ((3.13963 48.59977, 3.13951 48.59983, ...  4.874811e+07  
229  POLYGON ((1.83232 48.56349, 1.83260 48.56506, ...  4.634931e+07  
152  POLYGON ((2.23350 48.39672, 2.23293 48.39656, ...  4.596232e+07  
..                                                 ...           ...  
13   POLYGON ((2.41900 48.84939, 2.41900 48.84935, ...  9.057590e+05  
155  POLYGON ((1.90588 49.11131, 1.90559 49.11143, ...  7.725902e+05  
181  POLYGON ((2.29014 48.99680, 2.28979 48.99688, ...  7.250545e+05  
11   POLYGON ((2.41151 48.88322, 2.41148 48.88315, ...  7.027441e+05  
20   POLYGON ((2.48575 48.99933, 2.48549 48.99941, ...  9.722876e+04  

[1287 rows x 16 columns]

In [40]:
communes.rename(columns={'INSEE_COG': 'insee_com'}, inplace=True)

In [91]:
communes.head(n=10)

id    ID                       NOM  \
0  ARR_MUNI0000000009736045  None   Paris 3e Arrondissement   
1  ARR_MUNI0000000009736046  None   Paris 2e Arrondissement   
2  ARR_MUNI0000000009736545  None   Paris 4e Arrondissement   
3  ARR_MUNI0000000009736544  None   Paris 5e Arrondissement   
4  ARR_MUNI0000000009736543  None   Paris 6e Arrondissement   
5  ARR_MUNI0000000009736043  None   Paris 9e Arrondissement   
6  ARR_MUNI0000000009736042  None  Paris 10e Arrondissement   
7  ARR_MUNI0000000009736035  None  Paris 11e Arrondissement   
8  ARR_MUNI0000000009736531  None  Paris 13e Arrondissement   
9  ARR_MUNI0000000009736044  None   Paris 8e Arrondissement   

                      NOM_M INSEE_COM           STATUT  POPULATION INSEE_CAN  \
0   PARIS 3E ARRONDISSEMENT     75103  Capitale d'état       34025        NR   
1   PARIS 2E ARRONDISSEMENT     75102  Capitale d'état       21595        NR   
2   PARIS 4E ARRONDISSEMENT     75104  Capitale d'état       29131        NR   
3   PARIS 5E ARRONDISSEMENT     75105  Capitale d'état       58227        NR   
4   PARIS 6E ARRONDISSEMENT     75106  Capitale d'état       40303        NR   
5   PARIS 9E ARRONDISSEMENT     75109  Capitale d'état       60026        NR   
6  PARIS 10E ARRONDISSEMENT     75110  Capitale d'état       86472        NR   
7  PARIS 11E ARRONDISSEMENT     75111  Capitale d'état      145208        NR   
8  PARIS 13E ARRONDISSEMENT     75113  Capitale d'état      180005        NR   
9   PARIS 8E ARRONDISSEMENT     75108  Capitale d'état       35655        NR   

  INSEE_ARR INSEE_DEP INSEE_REG SIREN_EPCI                            source  \
0         1        75        11  200054781  IGN:EXPRESS-COG-CARTO-TERRITOIRE   
1         1        75        11  200054781  IGN:EXPRESS-COG-CARTO-TERRITOIRE   
2         1        75        11  200054781  IGN:EXPRESS-COG-CARTO-TERRITOIRE   
3         1        75        11  200054781  IGN:EXPRESS-COG-CARTO-TERRITOIRE   
4         1        75        11  200054781  IGN:EXPRESS-COG-CARTO-TERRITOIRE   
5         1        75        11  200054781  IGN:EXPRESS-COG-CARTO-TERRITOIRE   
6         1        75        11  200054781  IGN:EXPRESS-COG-CARTO-TERRITOIRE   
7         1        75        11  200054781  IGN:EXPRESS-COG-CARTO-TERRITOIRE   
8         1        75        11  200054781  IGN:EXPRESS-COG-CARTO-TERRITOIRE   
9         1        75        11  200054781  IGN:EXPRESS-COG-CARTO-TERRITOIRE   

  insee_com                                           geometry       surface  
0     75103  POLYGON ((2.35016 48.86199, 2.35019 48.86203, ...  1.170282e+06  
1     75102  POLYGON ((2.34792 48.87069, 2.34827 48.87062, ...  9.923087e+05  
2     75104  POLYGON ((2.36849 48.85580, 2.36873 48.85482, ...  1.596122e+06  
3     75105  POLYGON ((2.33666 48.83967, 2.33672 48.84011, ...  2.543140e+06  
4     75106  POLYGON ((2.33292 48.85934, 2.33339 48.85924, ...  2.152364e+06  
5     75109  POLYGON ((2.32589 48.86958, 2.32597 48.87009, ...  2.179290e+06  
6     75110  POLYGON ((2.36471 48.88437, 2.36527 48.88433, ...  2.890030e+06  
7     75111  POLYGON ((2.37690 48.87205, 2.37701 48.87195, ...  3.650811e+06  
8     75113  POLYGON ((2.34390 48.81577, 2.34394 48.81601, ...  7.138222e+06  
9     75108  POLYGON ((2.32717 48.88346, 2.32716 48.88323, ...  3.880853e+06

In [90]:
communes['INSEE_COM'] = communes['insee_com']

## Les vélibs en Ile de France

In [42]:
url2 = "https://opendata.paris.fr/explore/dataset/velib-emplacement-des-stations/download/?format=geojson&timezone=Europe/Berlin&lang=fr"

In [43]:
velib = gpd.read_file(url2)

In [44]:
velib.sample(n=15)

capacity                                   name stationcode  \
319         30    Cité des Sciences et de l'Industrie       19009   
685         39      Paul Vaillant Couturier - Chaptal       23001   
20          26                         Mahatma Gandhi       16123   
1247        23  Halle Freyssinet - Parvis Alan Turing       13051   
184         30         Charles Gide - Fort de Bicêtre       42701   
1422        17              Maine - Antoine Bourdelle       15002   
632         30                   Louis Ganne - Davout       20045   
1163        36                        Ecoles - Carmes        5007   
1182        34                  Paul Doumer - La Tour       16016   
304         23             Guy Môquet - Etienne Dolet       22406   
1070        55                 Parvis Corentin Celton       21311   
811         55            René Coty - Parc Montsouris       14016   
230         34             Chartres - La Charbonnière       18107   
749         28                Blanqui - Docteur Bauer       34013   
895         36          11 Novembre 1918 - 8 Mai 1945       45003   

                      geometry  
319   POINT (2.38460 48.89660)  
685   POINT (2.27753 48.89348)  
20    POINT (2.26421 48.87636)  
1247  POINT (2.37114 48.83285)  
184   POINT (2.35398 48.80719)  
1422  POINT (2.32026 48.84309)  
632   POINT (2.40973 48.86339)  
1163  POINT (2.34735 48.84890)  
1182  POINT (2.28084 48.86012)  
304   POINT (2.29808 48.81527)  
1070  POINT (2.27843 48.82747)  
811   POINT (2.33612 48.82479)  
230   POINT (2.35347 48.88476)  
749   POINT (2.34072 48.90774)  
895   POINT (2.53824 48.80890)

In [60]:
print(velib.shape)

(1471, 4)


In [138]:
gdf_velib = velib

In [139]:
gdf_velib.sample(10)

capacity                                  name stationcode  \
539         59                      Square Cambronne       15010   
684         26                               Beffroy       22003   
781         18       Gabriel Péri - Francisco Ferrer       33102   
1298        50         Lac Daumesnil - Saint-Maurice       12120   
119         22         Vincent Auriol - Louise Weiss       13049   
916         29  Vieille du Temple - Francs Bourgeois        4013   
1348        43        Jean Jaurès - Salvador Allende       22402   
436         27           Lucien Lanternier - Le Luth       22303   
1469        39                    Belfort - Voltaire       11020   
1044        29  Anatole France - President Roosevelt       33017   

                      geometry  
539   POINT (2.30257 48.84757)  
684   POINT (2.26139 48.88659)  
781   POINT (2.40734 48.88634)  
1298  POINT (2.41837 48.82775)  
119   POINT (2.37039 48.83400)  
916   POINT (2.35822 48.85826)  
1348  POINT (2.30266 48.82148)  
436   POINT (2.28898 48.93159)  
1469  POINT (2.38299 48.85634)  
1044  POINT (2.38141 48.91552)

On associe désormais aux stations vélib leur localisation (commune) pour créer deux nouvelles variables d'intérêt : la capacité vélib par commune (cap_com) et le nombre de stations vélib par communes (stat_com)

In [140]:
# Étape 1 : Spatial join pour associer à chaque station Vélib le code commune et le nom de la commune
gdf_velib_with_communes = gpd.sjoin(gdf_velib, communes[['id', 'NOM', 'INSEE_COM', 'geometry']], predicate='within')

print(gdf_velib_with_communes.shape)

(1472, 8)


Nous remplaçons un problème avec le code Insee des arrondissements parisiens qu'il faut corriger avant de calculer les indicatrices par arrondissement.

In [141]:
# Créer un dictionnaire de mapping des codes INSEE_COMMUNE pour les arrondissements de Paris
mapping_arrondissements = {
    'Paris 1er Arrondissement': '75101',
    'Paris 2e Arrondissement': '75102',
    'Paris 3e Arrondissement': '75103',
    'Paris 4e Arrondissement': '75104',
    'Paris 5e Arrondissement': '75105',
    'Paris 6e Arrondissement': '75106',
    'Paris 7e Arrondissement': '75107',
    'Paris 8e Arrondissement': '75108',
    'Paris 9e Arrondissement': '75109',
    'Paris 10e Arrondissement': '75110',
    'Paris 11e Arrondissement': '75111',
    'Paris 12e Arrondissement': '75112',
    'Paris 13e Arrondissement': '75113',
    'Paris 14e Arrondissement': '75114',
    'Paris 15e Arrondissement': '75115',
    'Paris 16e Arrondissement': '75116',
    'Paris 17e Arrondissement': '75117',
    'Paris 18e Arrondissement': '75118',
    'Paris 19e Arrondissement': '75119',
    'Paris 20e Arrondissement': '75120'
}

# Appliquer le mapping pour remplacer les codes INSEE_COMMUNE
gdf_velib_with_communes['INSEE_COM'] = gdf_velib_with_communes.apply(lambda row: mapping_arrondissements.get(row['NOM'], row['INSEE_COM']), axis=1)

# Afficher le DataFrame avec les codes INSEE_COMMUNE mis à jour
print(gdf_velib_with_communes)

      capacity                               name stationcode  \
0           21                 Toudouze - Clauzel        9020   
2           14  Abbeville - Faubourg Poissonnière        9002   
33          17                   Choron - Martyrs        9016   
39          22        Godot de Mauroy - Madeleine        9034   
78          24        Victoire - Chaussée d'Antin        9116   
...        ...                                ...         ...   
1159        31       Louis Pasteur - Albert Petit       22206   
1262        20               Stalingrad - Barbara       22208   
1347        28                 Tertres - Fontaine       22203   
1230        27                 Place Aimé Césaire       21402   
1309        33                  Mairie de Clamart       21401   

                      geometry  index_right                        id  \
0     POINT (2.33736 48.87930)            5  ARR_MUNI0000000009736043   
2     POINT (2.34915 48.87922)            5  ARR_MUNI0000000009736043   


In [142]:
# Étape 2 : Créer les variables cap_com et stat_com
df_aggregated = gdf_velib_with_communes.groupby('INSEE_COM').agg(
    cap_com=('capacity', 'sum'),
    stat_com=('index_right', 'count')
).reset_index()

print(df_aggregated.shape)

(82, 3)


In [143]:
df_aggregated.sample(10)

INSEE_COM  cap_com  stat_com
69     94033      327        11
17     75118     1798        61
35     92044      309        10
0      75101      710        26
5      75106     1028        34
74     94046      180         7
60     93064      103         4
57     93055      563        21
36     92046      238         8
11     75112     2656        67

In [188]:
# Étape 3 : Merge avec le GeoDataFrame des communes
df_velib = pd.merge(df_aggregated, communes, how='left', on='INSEE_COM')

In [189]:
print(df_velib.shape)

(82, 18)


In [190]:
# Liste des colonnes à supprimer
colonnes_a_supprimer = ['ID', 'id','INSEE_CAN', 'INSEE_ARR', 'SIREN_EPCI', 'source', 'insee_com']

# Supprimer les colonnes
df_velib = df_velib.drop(colonnes_a_supprimer, axis=1)

In [191]:
df_velib.head(10)

INSEE_COM  cap_com  stat_com                       NOM  \
0     75101      710        26  Paris 1er Arrondissement   
1     75102      747        25   Paris 2e Arrondissement   
2     75103      437        15   Paris 3e Arrondissement   
3     75104      712        26   Paris 4e Arrondissement   
4     75105     1169        36   Paris 5e Arrondissement   
5     75106     1028        34   Paris 6e Arrondissement   
6     75107     1060        27   Paris 7e Arrondissement   
7     75108     1560        51   Paris 8e Arrondissement   
8     75109     1118        45   Paris 9e Arrondissement   
9     75110     1619        55  Paris 10e Arrondissement   

                      NOM_M           STATUT  POPULATION INSEE_DEP INSEE_REG  \
0  PARIS 1ER ARRONDISSEMENT  Capitale d'état       15917        75        11   
1   PARIS 2E ARRONDISSEMENT  Capitale d'état       21595        75        11   
2   PARIS 3E ARRONDISSEMENT  Capitale d'état       34025        75        11   
3   PARIS 4E ARRONDISSEMENT  Capitale d'état       29131        75        11   
4   PARIS 5E ARRONDISSEMENT  Capitale d'état       58227        75        11   
5   PARIS 6E ARRONDISSEMENT  Capitale d'état       40303        75        11   
6   PARIS 7E ARRONDISSEMENT  Capitale d'état       48888        75        11   
7   PARIS 8E ARRONDISSEMENT  Capitale d'état       35655        75        11   
8   PARIS 9E ARRONDISSEMENT  Capitale d'état       60026        75        11   
9  PARIS 10E ARRONDISSEMENT  Capitale d'état       86472        75        11   

                                            geometry       surface  
0  POLYGON ((2.34464 48.85409, 2.34411 48.85428, ...  1.828408e+06  
1  POLYGON ((2.34792 48.87069, 2.34827 48.87062, ...  9.923087e+05  
2  POLYGON ((2.35016 48.86199, 2.35019 48.86203, ...  1.170282e+06  
3  POLYGON ((2.36849 48.85580, 2.36873 48.85482, ...  1.596122e+06  
4  POLYGON ((2.33666 48.83967, 2.33672 48.84011, ...  2.543140e+06  
5  POLYGON ((2.33292 48.85934, 2.33339 48.85924, ...  2.152364e+06  
6  POLYGON ((2.31671 48.84686, 2.31638 48.84677, ...  4.087721e+06  
7  POLYGON ((2.32717 48.88346, 2.32716 48.88323, ...  3.880853e+06  
8  POLYGON ((2.32589 48.86958, 2.32597 48.87009, ...  2.179290e+06  
9  POLYGON ((2.36471 48.88437, 2.36527 48.88433, ...  2.890030e+06

In [192]:
# Fichier intermédiaire :

gdf_velib = gpd.GeoDataFrame(df_velib, geometry='geometry')

gdf_velib.to_file('df_velib.geojson', driver='GeoJSON')


In [193]:
# df_velib_charge = gpd.read_file('df_velib.geojson')

# Afficher les premières lignes du GeoDataFrame chargé
# print(df_velib_charge.head())

## Concaténation des bases d'aménagements cyclables

In [194]:
print(df_velib.columns)

Index(['INSEE_COM', 'cap_com', 'stat_com', 'NOM', 'NOM_M', 'STATUT',
       'POPULATION', 'INSEE_DEP', 'INSEE_REG', 'geometry', 'surface'],
      dtype='object')


In [195]:
print(df_amenagements.columns)

Index(['nom_com', 'insee_com', 'longueur', 'longueur_pond_hw',
       'longueur_pond_adg', 'difference_hw', 'difference_adg'],
      dtype='object')


In [196]:
# Fusion des DataFrames
df_velo = pd.merge(df_amenagements, df_velib, left_on=['insee_com'], right_on =['INSEE_COM'], how='left')

In [197]:
print(df_velo.shape)

(1005, 18)


In [198]:
df_velo.head(15)

nom_com insee_com  longueur  longueur_pond_hw  \
0   10e Arrondissement     75110     60281          55764.25   
1   11e Arrondissement     75111     90116          84508.75   
2   12e Arrondissement     75112    201748         201066.75   
3   13e Arrondissement     75113    137731         128770.50   
4   14e Arrondissement     75114    102618          94183.00   
5   15e Arrondissement     75115    148909         136007.00   
6   16e Arrondissement     75116    228240         226273.25   
7   17e Arrondissement     75117    114273         108548.00   
8   18e Arrondissement     75118    118171         113091.00   
9   19e Arrondissement     75119    133478         125675.25   
10  1er Arrondissement     75101     40320          36750.75   
11  20e Arrondissement     75120    112161         104438.00   
12   2e Arrondissement     75102     25006          21323.25   
13   3e Arrondissement     75103     26150          24104.50   
14   4e Arrondissement     75104     42547          39476.75   

    longueur_pond_adg  difference_hw  difference_adg INSEE_COM  cap_com  \
0            51170.00       -4516.75        -9111.00     75110   1619.0   
1            73843.00       -5607.25       -16273.00     75111   2126.0   
2           202496.25        -681.25          748.25     75112   2656.0   
3           119280.50       -8960.50       -18450.50     75113   2256.0   
4            87486.75       -8435.00       -15131.25     75114   1830.0   
5           128724.50      -12902.00       -20184.50     75115   3345.0   
6           231651.00       -1966.75         3411.00     75116   2129.0   
7           100289.50       -5725.00       -13983.50     75117   2085.0   
8            98709.50       -5080.00       -19461.50     75118   1798.0   
9           128372.00       -7802.75        -5106.00     75119   1762.0   
10           34558.00       -3569.25        -5762.00     75101    710.0   
11           90906.25       -7723.00       -21254.75     75120   1886.0   
12           16419.25       -3682.75        -8586.75     75102    747.0   
13           19149.50       -2045.50        -7000.50     75103    437.0   
14           35427.50       -3070.25        -7119.50     75104    712.0   

    stat_com                       NOM                     NOM_M  \
0       55.0  Paris 10e Arrondissement  PARIS 10E ARRONDISSEMENT   
1       57.0  Paris 11e Arrondissement  PARIS 11E ARRONDISSEMENT   
2       67.0  Paris 12e Arrondissement  PARIS 12E ARRONDISSEMENT   
3       68.0  Paris 13e Arrondissement  PARIS 13E ARRONDISSEMENT   
4       55.0  Paris 14e Arrondissement  PARIS 14E ARRONDISSEMENT   
5       90.0  Paris 15e Arrondissement  PARIS 15E ARRONDISSEMENT   
6       65.0  Paris 16e Arrondissement  PARIS 16E ARRONDISSEMENT   
7       63.0  Paris 17e Arrondissement  PARIS 17E ARRONDISSEMENT   
8       61.0  Paris 18e Arrondissement  PARIS 18E ARRONDISSEMENT   
9       61.0  Paris 19e Arrondissement  PARIS 19E ARRONDISSEMENT   
10      26.0  Paris 1er Arrondissement  PARIS 1ER ARRONDISSEMENT   
11      71.0  Paris 20e Arrondissement  PARIS 20E ARRONDISSEMENT   
12      25.0   Paris 2e Arrondissement   PARIS 2E ARRONDISSEMENT   
13      15.0   Paris 3e Arrondissement   PARIS 3E ARRONDISSEMENT   
14      26.0   Paris 4e Arrondissement   PARIS 4E ARRONDISSEMENT   

             STATUT  POPULATION INSEE_DEP INSEE_REG  \
0   Capitale d'état     86472.0        75        11   
1   Capitale d'état    145208.0        75        11   
2   Capitale d'état    139297.0        75        11   
3   Capitale d'état    180005.0        75        11   
4   Capitale d'état    135592.0        75        11   
5   Capitale d'état    230981.0        75        11   
6   Capitale d'état    165523.0        75        11   
7   Capitale d'état    166543.0        75        11   
8   Capitale d'état    192468.0        75        11   
9   Capitale d'état    184573.0        75        11   
10  Capitale d'état     15917.0        75        11   
11  Capitale d'état    194994.0        7

In [199]:
print(df_velo.columns)

Index(['nom_com', 'insee_com', 'longueur', 'longueur_pond_hw',
       'longueur_pond_adg', 'difference_hw', 'difference_adg', 'INSEE_COM',
       'cap_com', 'stat_com', 'NOM', 'NOM_M', 'STATUT', 'POPULATION',
       'INSEE_DEP', 'INSEE_REG', 'geometry', 'surface'],
      dtype='object')


In [200]:
columns_to_drop = ['INSEE_COM', 'NOM', 'NOM_M', 'STATUT', 'POPULATION', 'INSEE_DEP', 'INSEE_REG']

df_velo = df_velo.drop(columns=columns_to_drop)


In [201]:
missing_values = df_velo.isna().sum()

print(missing_values)

nom_com                0
insee_com              0
longueur               0
longueur_pond_hw       0
longueur_pond_adg      0
difference_hw          0
difference_adg         0
cap_com              923
stat_com             923
geometry             923
surface              923
dtype: int64


In [202]:
df_velo = pd.merge(df_velo, communes, on='insee_com', how='left')

In [203]:
df_velo['geometry'] = df_velo['geometry_y']
df_velo['surface'] = df_velo['surface_y']
df_velo['population'] = df_velo['POPULATION']

In [204]:
columns_to_drop = ['POPULATION', 'id', 'ID','INSEE_CAN', 'INSEE_ARR', 'INSEE_COM', 'NOM', 'NOM_M', 'STATUT', 'INSEE_DEP', 'INSEE_REG', 'SIREN_EPCI', 'geometry_y', 'geometry_x', 'surface_x', 'surface_y']

df_velo = df_velo.drop(columns=columns_to_drop)


In [205]:
print(df_velo.shape)

(1005, 13)


Point d'étape : nous enregistrons cette base intermédiaire des vélos pour ne pas avoir à refaire toutes les étapes toujours.

In [206]:
# Sauvegarder le DataFrame au format GeoJSON

gdf_velo = gpd.GeoDataFrame(df_velo, geometry='geometry')

# Sauvegarder le GeoDataFrame au format GeoJSON
gdf_velo.to_file('df_velo.geojson', driver='GeoJSON')

## Données socio-démographiques sur l'Ile de France

## Données complémentaires : accidents de la route, transports écologiques alternatifs (réseau ferré, bus)

Nous utilisons la [base de données anuuelles des accidents corporels de la circulation routière entre 2005 et 2022](https://www.data.gouv.fr/fr/datasets/bases-de-donnees-annuelles-des-accidents-corporels-de-la-circulation-routiere-annees-de-2005-a-2022/). Les deux variables explicatives que nous souhaitons construire sont les suivantes : nombre d'accident par communes d'intérêt sur la période, et nombre de blessés (graves), dans la mesure où nous faisons l'hypothèse que les accidents graves sont plus pris en compte par les pouvoirs publics et l'opinion des habitants.

In [28]:
# Chargez les bases de données caractéristiques et usagers
caracteristiques = pd.read_csv('caracteristiques-2022.csv', delimiter=';')
usagers = pd.read_csv('usagers-2022.csv', delimiter=';')

In [29]:
usagers.head(n=10)

Num_Acc  id_usager id_vehicule num_veh  place  catu  grav  sexe  \
0  202200000001  1 099 700     813 952     A01      1     1     3     1   
1  202200000001  1 099 701     813 953     B01      1     1     1     1   
2  202200000002  1 099 698     813 950     B01      1     1     4     1   
3  202200000002  1 099 699     813 951     A01      1     1     1     1   
4  202200000003  1 099 696     813 948     A01      1     1     1     1   
5  202200000003  1 099 697     813 949     B01      1     1     4     2   
6  202200000004  1 099 694     813 947     A01      1     1     1     2   
7  202200000004  1 099 695     813 947     A01     10     3     4     2   
8  202200000005  1 099 692     813 945     B01      1     1     1     2   
9  202200000005  1 099 693     813 946     A01      1     1     4     2   

   an_nais  trajet  secu1  secu2  secu3  locp actp  etatp  
0   2008.0       5      2      8     -1    -1   -1     -1  
1   1948.0       5      1      8     -1    -1   -1     -1  
2   1988.0       9      1      0     -1     0    0     -1  
3   1970.0       4      1      0     -1     0    0     -1  
4   2002.0       0      1      0     -1    -1   -1     -1  
5   1987.0       9      1      0     -1    -1   -1     -1  
6   2000.0       0      1      8     -1    -1   -1     -1  
7   1967.0       5      0     -1     -1     2    3      1  
8   1975.0       0      1      0     -1    -1   -1     -1  
9   1996.0       0      1      0     -1    -1   -1     -1

In [63]:
caracteristiques.shape

(55302, 15)

In [31]:
caracteristiques = caracteristiques.rename(columns={'Accident_Id': 'Num_Acc'})

In [32]:
# Aggrégation des informations par accident

# Créer les colonnes 'mort', 'hosp', 'indemne' et 'leger' en fonction de la gravité
usagers['mort'] = (usagers['grav'] == 2).astype(int)
usagers['hosp'] = (usagers['grav'] == 3).astype(int)
usagers['indemne'] = (usagers['grav'] == 1).astype(int)
usagers['leger'] = (usagers['grav'] == 4).astype(int)

In [33]:
# Pivoter les données pour obtenir le décompte par catégorie de gravité
accident_counts = usagers.groupby('Num_Acc').agg({
    'indemne': 'sum',
    'mort': 'sum',
    'hosp': 'sum',
    'leger': 'sum'
}).reset_index()

In [34]:
accident_counts.head(10)

Num_Acc  indemne  mort  hosp  leger
0  202200000001        1     0     1      0
1  202200000002        1     0     0      1
2  202200000003        1     0     0      1
3  202200000004        1     0     0      1
4  202200000005        1     0     0      1
5  202200000006        1     0     0      1
6  202200000007        1     0     0      2
7  202200000008        1     0     0      1
8  202200000009        1     0     0      1
9  202200000010        1     0     0      1

In [48]:
# Fusionnez les deux bases de données sur la colonne 'Num_Acc'
accidents = pd.merge(caracteristiques, usagers, on='Num_Acc', how='inner')

In [54]:
# On ne s'intéresse qu'à l'Ile de France

accidents_selection = accidents[['Num_Acc', 'an', 'dep', 'com', 'agg', 'int', 'lat', 'long', 'mort', 'hosp', 'indemne', 'leger']]

In [55]:
# Imprimer les départements uniques dans la base sélectionnée
print(accidents_selection['dep'].unique())

['26' '25' '22' '16' '13' '12' '10' '09' '06' '03' '02' '75' '94' '93'
 '92' '76' '68' '972' '63' '62' '51' '2B' '65' '33' '71' '89' '61' '29'
 '38' '14' '60' '45' '73' '53' '24' '27' '17' '48' '973' '77' '67' '988'
 '08' '56' '69' '91' '78' '64' '59' '49' '42' '37' '34' '32' '31' '11'
 '35' '47' '86' '01' '74' '79' '2A' '46' '39' '72' '05' '18' '19' '971'
 '987' '66' '84' '85' '95' '57' '54' '52' '50' '44' '974' '83' '28' '15'
 '07' '80' '30' '41' '82' '55' '43' '40' '04' '986' '88' '36' '87' '21'
 '976' '81' '58' '70' '90' '23' '977' '978' '975']


In [56]:
# Trier par département
accidents_selection = accidents_selection.sort_values(by='dep')

In [59]:
# Ne conserver que les enregistrements liés à l'Île-de-France
departements_idf = ["75", "77", "78", "91", "92","93", "94", "95"]
accidents_idf = accidents_selection[accidents_selection['dep'].isin(departements_idf)]

In [61]:
accidents_idf.head(n=10)

Num_Acc    an dep    com  agg  int            lat          long  \
94161  202200041119  2022  75  75117    2    1  48,8779706984  2,2873992231   
25938  202200011451  2022  75  75101    2    2  48,8693990000  2,3251790000   
6484   202200002865  2022  75  75120    2    2  48,8505070000  2,4064080000   
6485   202200002866  2022  75  75120    2    2  48,8645560000  2,4073410000   
94160  202200041118  2022  75  75113    2    1  48,8161200000  2,3599110000   
94162  202200041119  2022  75  75117    2    1  48,8779706984  2,2873992231   
25931  202200011448  2022  75  75105    2    3  48,8447930000  2,3547480000   
6472   202200002861  2022  75  75114    2    1  48,8237130000  2,3089650000   
8789   202200003872  2022  75  75118    2    1  48,9006169404  2,3317149368   
25939  202200011452  2022  75  75119    2    1  48,8798310000  2,3704110000   

       mort  hosp  indemne  leger  
94161     0     0        0      1  
25938     0     0        1      0  
6484      0     0        0      1  
6485      0     0        0      1  
94160     0     0        0      1  
94162     0     0        1      0  
25931     0     0        0      1  
6472      0     0        1      0  
8789      0     0        1      0  
25939     0     0        0      1

In [62]:
accidents_idf.shape

(37427, 12)

In [64]:
# Grouper par département (dep) et commune (com) et calculer les sommes
accidents_com_sum = accidents_idf.groupby(['dep', 'com']).agg({
    'mort': 'sum',
    'hosp': 'sum',
    'indemne': 'sum',
    'leger': 'sum',
    'Num_Acc': 'count'  # Nombre d'accidents par commune
}).reset_index()

# Renommer les colonnes
accidents_com_sum = accidents_com_sum.rename(columns={
    'mort': 'mort_com',
    'hosp': 'hosp_com',
    'indemne': 'indemne_com',
    'leger': 'leger_com',
    'Num_Acc': 'acc_com'
})

In [65]:
# Afficher la base résultante
print(accidents_com_sum)

    dep    com  mort_com  hosp_com  indemne_com  leger_com  acc_com
0    75  75101         1         3          108        143      256
1    75  75102         2         3           65         92      162
2    75  75103         0         7           91        101      200
3    75  75104         0         5          138        173      316
4    75  75105         0         9          123        136      269
..   ..    ...       ...       ...          ...        ...      ...
723  95  95675         0         0            1          2        3
724  95  95678         0         2           16          9       27
725  95  95680         0         5           17         29       51
726  95  95682         0         0            3          6        9
727  95  95690         0         0            0          1        1

[728 rows x 7 columns]


Nous avons donc les informations des accidents pour 728 communes d'Ile de France, nous pouvons enregistrer une base intermédiaire (df_accidents) avant de l'associer à la base des aménagements cyclables.

In [66]:
accidents_com_sum.to_csv('df_accidents.csv', index=False)

### Données de l'INSEE 

In [111]:
# URL du fichier zip complet de l'insee
url3 = "https://www.insee.fr/fr/statistiques/fichier/2521169/base_cc_comparateur_xlsx.zip"

In [118]:
# Télécharger le fichier zip
# response = requests.get(url3)
# zip_file = ZipFile(BytesIO(response.content))

In [119]:
# Liste des fichiers dans le zip
# file_list = zip_file.namelist()
# print(file_list)

['base_cc_comparateur.xlsx']


In [120]:
# Nom du fichier à extraire
# insee_file_name = "base_cc_comparateur.xlsx"

In [121]:
# Extraire les données de l'Île-de-France
# with zip_file.open(insee_file_name) as file:
    # Charger le CSV en DataFrame
 #    df = pd.read_excel(file, skiprows=5)
        
# Problème de fichier corrompu

ValueError: Unable to read workbook: could not read stylesheet from base_cc_comparateur.xlsx.
This is most probably because the workbook source files contain some invalid XML.
Please see the exception for more details.

In [124]:
df_INSEE = pd.read_excel("base_cc_comparateur.xlsx", skiprows=5)



In [125]:
print(df_INSEE)

      CODGEO                   LIBGEO  REG  DEP  P20_POP  P14_POP  SUPERF  \
0      01001  L'Abergement-Clémenciat   84   01    806.0    767.0   15.95   
1      01002    L'Abergement-de-Varey   84   01    262.0    239.0    9.15   
2      01004        Ambérieu-en-Bugey   84   01  14288.0  14022.0   24.60   
3      01005      Ambérieux-en-Dombes   84   01   1782.0   1627.0   15.92   
4      01006                  Ambléon   84   01    113.0    109.0    5.88   
...      ...                      ...  ...  ...      ...      ...     ...   
34940  97613            M'Tsangamouji    6  976      NaN      NaN     NaN   
34941  97614                 Ouangani    6  976      NaN      NaN     NaN   
34942  97615                 Pamandzi    6  976      NaN      NaN     NaN   
34943  97616                     Sada    6  976      NaN      NaN     NaN   
34944  97617                 Tsingoni    6  976      NaN      NaN     NaN   

       NAIS1420  DECE1420      P20_MEN  ...  P20_ACT1564  ETTOT21  ETAZ21  

In [128]:
# Liste des départements à conserver
departements_a_conserver = ["75", "77", "78", "91","92", "93", "94", "95"]

# Créer un masque booléen
masque_departements = df_INSEE['DEP'].isin(departements_a_conserver)

# Appliquer le masque pour sélectionner les lignes
df_INSEE_idf = df_INSEE[masque_departements]

In [130]:
print(df_INSEE_idf)

      CODGEO               LIBGEO  REG DEP    P20_POP    P14_POP  SUPERF  \
29252  75056                Paris   11  75  2145906.0  2220445.0  105.40   
29961  77001     Achères-la-Forêt   11  77     1151.0     1177.0   12.60   
29962  77002              Amillis   11  77      819.0      808.0   20.06   
29963  77003           Amponville   11  77      342.0      376.0   12.30   
29964  77004             Andrezel   11  77      299.0      285.0    8.08   
...      ...                  ...  ...  ..        ...        ...     ...   
34811  95676   Villers-en-Arthies   11  95      490.0      503.0    8.25   
34812  95678        Villiers-Adam   11  95      857.0      838.0    9.82   
34813  95680      Villiers-le-Bel   11  95    28636.0    27571.0    7.30   
34814  95682      Villiers-le-Sec   11  95      193.0      177.0    3.26   
34815  95690  Wy-dit-Joli-Village   11  95      330.0      318.0    8.37   

       NAIS1420  DECE1420       P20_MEN  ...   P20_ACT1564   ETTOT21  ETAZ21  \
29252  

Suivant le [dictionnaire des variables](https://www.insee.fr/fr/statistiques/2521169#dictionnaire), nous conservons les variables suivantes : CODGEO, LIBGEO, SUPERF, P20_RP (Nombre de résidences principales en 2020), MED20 (Médiane du niveau de vie en 2020), TP6020 (Taux de pauvreté en 2020), P20_EMPLT (Nombre d'emplois au lieu de travail en 2020), ETOQ21 (Établissements actifs de l'administration publique, enseignement, santé et action sociale fin 2021).

In [138]:
# Liste des variables à conserver
variables_a_conserver = ['CODGEO', 'SUPERF', 'P20_RP', 'MED20', 'TP6020', 'P20_EMPLT', 'ETOQ21']

# Sélectionner uniquement les colonnes spécifiées
df_INSEE_idf = df_INSEE_idf[variables_a_conserver]

In [139]:
print(df_INSEE_idf.head())

      CODGEO  SUPERF        P20_RP  MED20 TP6020     P20_EMPLT   ETOQ21
29252  75056  105.40  1.131610e+06  28790   15,0  1.865785e+06  12258.0
29961  77001   12.60  4.760000e+02  29170      s  1.725137e+02      6.0
29962  77002   20.06  3.249246e+02  24800      s  2.674927e+02      6.0
29963  77003   12.30  1.494284e+02  27690      s  4.469306e+01      3.0
29964  77004    8.08  1.257745e+02  27420      s  3.999061e+01      2.0


In [140]:
df_INSEE_idf.to_csv('df_insee.csv', index=False)

## Base finale

In [165]:
# Charger dataframe df_velo

df_velo = gpd.read_file('df_velo.geojson')

df_velo.head(10)

nom_com insee_com  longueur  longueur_pond_hw  \
0  10e Arrondissement     75110     60281          55764.25   
1  11e Arrondissement     75111     90116          84508.75   
2  12e Arrondissement     75112    201748         201066.75   
3  13e Arrondissement     75113    137731         128770.50   
4  14e Arrondissement     75114    102618          94183.00   
5  15e Arrondissement     75115    148909         136007.00   
6  16e Arrondissement     75116    228240         226273.25   
7  17e Arrondissement     75117    114273         108548.00   
8  18e Arrondissement     75118    118171         113091.00   
9  19e Arrondissement     75119    133478         125675.25   

   longueur_pond_adg  difference_hw  difference_adg  cap_com  stat_com  \
0           51170.00       -4516.75        -9111.00   1619.0      55.0   
1           73843.00       -5607.25       -16273.00   2126.0      57.0   
2          202496.25        -681.25          748.25   2656.0      67.0   
3          119280.50       -8960.50       -18450.50   2256.0      68.0   
4           87486.75       -8435.00       -15131.25   1830.0      55.0   
5          128724.50      -12902.00       -20184.50   3345.0      90.0   
6          231651.00       -1966.75         3411.00   2129.0      65.0   
7          100289.50       -5725.00       -13983.50   2085.0      63.0   
8           98709.50       -5080.00       -19461.50   1798.0      61.0   
9          128372.00       -7802.75        -5106.00   1762.0      61.0   

                             source       surface  population  \
0  IGN:EXPRESS-COG-CARTO-TERRITOIRE  2.890030e+06       86472   
1  IGN:EXPRESS-COG-CARTO-TERRITOIRE  3.650811e+06      145208   
2  IGN:EXPRESS-COG-CARTO-TERRITOIRE  1.638084e+07      139297   
3  IGN:EXPRESS-COG-CARTO-TERRITOIRE  7.138222e+06      180005   
4  IGN:EXPRESS-COG-CARTO-TERRITOIRE  5.613081e+06      135592   
5  IGN:EXPRESS-COG-CARTO-TERRITOIRE  8.472060e+06      230981   
6  IGN:EXPRESS-COG-CARTO-TERRITOIRE  1.640971e+07      165523   
7  IGN:EXPRESS-COG-CARTO-TERRITOIRE  5.665304e+06      166543   
8  IGN:EXPRESS-COG-CARTO-TERRITOIRE  5.995586e+06      192468   
9  IGN:EXPRESS-COG-CARTO-TERRITOIRE  6.784861e+06      184573   

                                            geometry  
0  POLYGON ((2.36471 48.88437, 2.36527 48.88433, ...  
1  POLYGON ((2.37690 48.87205, 2.37701 48.87195, ...  
2  POLYGON ((2.39007 48.82569, 2.38909 48.82635, ...  
3  POLYGON ((2.34390 48.81577, 2.34394 48.81601, ...  
4  POLYGON ((2.34209 48.83834, 2.34208 48.83830, ...  
5  POLYGON ((2.26279 48.83392, 2.26356 48.83464, ...  
6  POLYGON ((2.27995 48.87857, 2.28010 48.87852, ...  
7  POLYGON ((2.32742 48.88351, 2.32734 48.88349, ...  
8  POLYGON ((2.37030 48.90185, 2.37025 48.90177, ...  
9  POLYGON ((2.37690 48.87205, 2.37537 48.87343, ...

In [166]:
# Charger dataframe df_accidents

df_accidents = pd.read_csv('df_accidents.csv')

df_accidents.head(10)

dep    com  mort_com  hosp_com  indemne_com  leger_com  acc_com
0   75  75101         1         3          108        143      256
1   75  75102         2         3           65         92      162
2   75  75103         0         7           91        101      200
3   75  75104         0         5          138        173      316
4   75  75105         0         9          123        136      269
5   75  75106         1         7          121        122      251
6   75  75107         2        13          173        225      413
7   75  75108         1        22          288        287      598
8   75  75109         0         8          123        142      273
9   75  75110         3        11          207        205      427

In [167]:
# Charger dataframe df_insee

df_insee = pd.read_csv('df_insee.csv')

df_insee.head(10)

CODGEO  SUPERF        P20_RP  MED20 TP6020     P20_EMPLT   ETOQ21
0   75056  105.40  1.131610e+06  28790   15,0  1.865785e+06  12258.0
1   77001   12.60  4.760000e+02  29170      s  1.725137e+02      6.0
2   77002   20.06  3.249246e+02  24800      s  2.674927e+02      6.0
3   77003   12.30  1.494284e+02  27690      s  4.469306e+01      3.0
4   77004    8.08  1.257745e+02  27420      s  3.999061e+01      2.0
5   77005   13.19  1.272534e+03  27940      s  5.288698e+02      8.0
6   77006   15.08  3.834529e+02  30530      s  2.474246e+02      7.0
7   77007    2.57  1.405743e+02  27730      s  2.428085e+01      1.0
8   77008    7.27  3.658635e+02  20340      s  4.481301e+01      2.0
9   77009   11.31  5.160000e+01  23030      s  3.672255e+01      1.0

In [170]:
df_INSEE_idf['CODGEO'] = df_INSEE_idf['CODGEO'].astype(int)
df_velo['CODGEO'] = df_velo['insee_com'].astype(int)
df_accidents['CODGEO'] = df_accidents['com'].astype(int)

print(type(df_INSEE_idf['CODGEO']))
print(type(df_velo['CODGEO']))
print(type(df_accidents['CODGEO']))


<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>


In [171]:
# Fusionner df_INSEE_idf et df_velo sur le code commune

df_base_temp = pd.merge(df_INSEE_idf, df_velo, on='CODGEO', how='right')

df_base_temp.sample(15)

CODGEO  SUPERF        P20_RP  MED20 TP6020     P20_EMPLT  ETOQ21  \
731    78576   10.15    651.761599  32530      s    209.608234     4.0   
1252   95627    7.60    239.862692  31890      s     92.163010     5.0   
441    77462   12.07    313.894449  23730      s    203.753064     2.0   
1188   95371    8.62   2111.300814  26090    7,0   2172.086345    14.0   
679    78440   11.99  11559.173000  17960   25,0  12543.105533   101.0   
233    77245   20.29    172.569142  24700      s     51.303826     1.0   
797    91106    4.68    524.996552  27110      s    145.497271     4.0   
1263   95676    8.25    204.893576  28010      s     79.773948     2.0   
1257   95651    4.30    418.371733  25460      s    127.200900     4.0   
332    77348    3.81     99.882660  27810      s    141.850883     1.0   
258    77270   13.57    346.164986  26820      s    159.562246     5.0   
77     77081    3.97    276.756391  26050      s    135.324284     2.0   
1068   94060    9.16   4372.448938  24450   10,0   2490.335764    26.0   
263    77275    5.38     52.454712  23890      s     14.013491     1.0   
798    91109    8.65    473.372466  25840      s    817.728428     3.0   

                  nom_com insee_com  longueur  longueur_pond_hw  \
731   Saint-Rémy-l'Honoré     78576    3779.0           3726.75   
1252        Vallangoujard     95627    1184.0           1336.75   
441                Thieux     77462     784.0            588.00   
1188       Marly-la-Ville     95371    1224.0           1076.00   
679           Les Mureaux     78440   75430.0          78528.75   
233                   NaN       NaN       NaN               NaN   
797            Breux-Jouy     91106     913.0            684.75   
1263   Villers-en-Arthies     95676     939.0            749.50   
1257             Vétheuil     95651   10482.0           9758.00   
332                   NaN       NaN       NaN               NaN   
258                   NaN       NaN       NaN               NaN   
77                    NaN       NaN       NaN               NaN   
1068     La Queue-en-Brie     94060   26262.0          28205.25   
263                   NaN       NaN       NaN               NaN   
798                   NaN       NaN       NaN               NaN   

      longueur_pond_adg  difference_hw  difference_adg  cap_com  stat_com  \
731             4606.00         -52.25          827.00      NaN       NaN   
1252            1336.75         152.75          152.75      NaN       NaN   
441              784.00        -196.00            0.00      NaN       NaN   
1188            1341.25        -148.00          117.25      NaN       NaN   
679            76389.75        3098.75          959.75      NaN       NaN   
233                 NaN            NaN             NaN      NaN       NaN   
797              913.00        -228.25            0.00      NaN       NaN   
1263             984.25        -189.50           45.25      NaN       NaN   
1257           10464.25        -724.00          -17.75      NaN       NaN   
332                 NaN            NaN             NaN      NaN       NaN   
258                 NaN            NaN             NaN      NaN       NaN   
77                  NaN            NaN             NaN      NaN       NaN   
1068           30393.00        1943.25         4131.00      NaN       NaN   
263                 NaN            NaN             NaN      NaN       NaN   
798                 NaN            NaN             NaN      NaN       NaN   

                                source       surface  population  \
731   IGN:EXPRESS-COG-CARTO-TERRITOIRE  1.024858e+07      1621.0   
1252  IGN:EXPRESS-COG-CARTO-TERRITOIRE  7.571426e+06       598.0   
441   IGN:EXPRESS-COG-CARTO-TERRITOIRE  1.207595e+07       872.0   
1188  IGN:EXPRESS-COG-CARTO-TERRITOIRE  8.782432e+06      5663.0   
679   IGN:EXPRESS-COG-CARTO-TERRITOIRE  1.238361e+07     33203.0   
233                                NaN           NaN         NaN   
797   IGN:EXPRESS-COG-CARTO-TERRITOIRE 

In [173]:
# Fusionner df_base_temp et df_accidents sur le code commune (CODGEO)
df_base = pd.merge(df_base_temp, df_accidents, on='CODGEO', how='left')

df_base.shape

(1268, 27)

In [174]:
df_base.to_csv('base_finale.csv', index=False)